### PR TITLE
[Feature-15148][dolphinscheduler-alert-plugins] add alert plugin aliyun-voice 

### LINKED ISSUE #15170: [Feature][dolphinscheduler-alert-plugins] add alert plugin aliyun-voice
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

 add a alert plugin aliyun-voice

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by xujiaqiang**
- close #15170

### CODE REVIEW

**Review by songjianet — CHANGES_REQUESTED** [review]
If there are no dependency changes, please do not submit the pnpm-lock file.

**Comment by xujiaqiang** [comment]
> 如果没有依赖项更改，请不要提交 pnpm-lock 文件。

rolled back code to origin/dev branch

**Comment by xujiaqiang** [comment]
> If there are no dependency changes, please do not submit the pnpm-lock file.

rolled back code to origin/dev branch

**Review by songjianet — DISMISSED** [review]

**Comment by qingwli** [comment]
Please create an issue and bind the pr to issue thanks. And please add the doc about the new plugin

**Comment by xujiaqiang** [comment]
> ## 拉取请求的目的
> ## 简要变更日志
> ## 验证此拉取请求
> 此拉取请求是代码清理，没有任何测试覆盖率。
> 
> _（或者）_
> 
> 此拉取请求已被现有测试覆盖，例如_（请描述测试）_。
> 
> （或者）
> 
> 此更改添加了测试，可以验证如下：
> 
> （或者）
> 
> 如果您的拉取请求包含不兼容的更改，您还应该将其添加到`docs/docs/en/guide/upgrede/incompatible.md`

> [Feature][dolphinscheduler-alert-plugins] add alert plugin aliyun-voice #15170

**Comment by xujiaqiang** [comment]
> 请创建一个问题并将 pr 绑定到问题，谢谢。请添加有关新插件的文档
completed

**Comment by xujiaqiang** [comment]
> 请创建一个问题并将 pr 绑定到问题，谢谢。请添加有关新插件的文档

> completed

**Comment by xujiaqiang** [comment]
> Please create an issue and bind the pr to issue thanks. And please add the doc about the new plugin

> completed

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceAlertChannel.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        param.setTtsCode(ttsCode);
+        VoiceParam.Connection connection = new VoiceParam.Connection();
+        String address = paramsMap.get(NAME_ADDRESS);
+        String accessKeyId = paramsMap.get(NAME_ACCESS_KEY_ID);
+        String accessKeySecret = paramsMap.get(NAME_ACCESS_KEY_SECRET);
+        connection.setAddress(address);
+        connection.setAccessKeyId(accessKeyId);
+        connection.setAccessKeySecret(accessKeySecret);
+        param.setConnection(connection);
+        String outId = String.valueOf((int) (1000 + Math.random() * (99999999 - 1 - 10000000)));
```
Comment (ruanwenjun) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceAlertChannel.java]: The outId is strange, is there doc about this param?

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/pom.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.dolphinscheduler</groupId>
+            <artifactId>dolphinscheduler-alert-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.aliyun</groupId>
+            <artifactId>dyvmsapi20170525</artifactId>
+        </dependency>
```
Comment (ruanwenjun) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/pom.xml:38]: Is seems you didn't define the version?

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceAlertChannel.java (hunk 1 above)
Comment (xujiaqiang) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceAlertChannel.java]: callback ID  , no practical significance , the ID reserved for the caller to use，

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/pom.xml (hunk 2 above)
Comment (xujiaqiang) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/pom.xml:38]: added

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceAlertChannel.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public final class VoiceAlertChannel implements AlertChannel {
+
+    @Override
+    public AlertResult process(AlertInfo info) {
+
+        AlertData alert = info.getAlertData();
```
[the remark this answers is not in the document]
Comment (xujiaqiang) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceAlertChannel.java]: completed

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceAlertChannel.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        Map<String, String> paramsMap = info.getAlertParams();
+        if (null == paramsMap) {
+            return new AlertResult("false", "aliyun-voice params is null");
+        }
+        VoiceParam voiceParam = buildVoiceParam(paramsMap);
+        VoiceSender voiceSender = new VoiceSender(voiceParam);
+
+        AlertResult alertResult = voiceSender.send();
+        Boolean flag = Boolean.parseBoolean(String.valueOf(alertResult.getStatus()));
```
[the remark this answers is not in the document]
Comment (xujiaqiang) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceAlertChannel.java]: completed

**Comment by zhongjiajie** [comment]
approval ci run.

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceSender.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        try {
+            Client client = createClient(voiceParam.getConnection());
+            SingleCallByTtsRequest singleCallByTtsRequest = new SingleCallByTtsRequest()
+                    .setCalledNumber(voiceParam.getCalledNumber())
+                    .setTtsCode(voiceParam.getTtsCode())
+                    .setOutId(voiceParam.getOutId());
+            RuntimeOptions runtime = new RuntimeOptions();
+            SingleCallByTtsResponse response = client.singleCallByTtsWithOptions(singleCallByTtsRequest, runtime);
+            if (response == null) {
+                alertResult.setMessage("vocie send error.");
```
Comment (qingwli) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceSender.java]: Please input real error reason in `message`

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceSender.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                return alertResult;
+            }
+            SingleCallByTtsResponseBody body = response.getBody();
+            alertResult.setStatus(body.code.toLowerCase().equals("ok") ? "true" : "false");
+        } catch (Exception e) {
+            log.error("vocie send error.", e);
+            alertResult.setMessage("vocie send error.");
```
Comment (qingwli) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceSender.java]: Ditto

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceSender.java [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+                    .setTtsCode(voiceParam.getTtsCode())
+                    .setOutId(voiceParam.getOutId());
+            RuntimeOptions runtime = new RuntimeOptions();
+            SingleCallByTtsResponse response = client.singleCallByTtsWithOptions(singleCallByTtsRequest, runtime);
+            if (response == null) {
+                alertResult.setMessage("vocie send error.");
+                return alertResult;
+            }
+            SingleCallByTtsResponseBody body = response.getBody();
+            alertResult.setStatus(body.code.toLowerCase().equals("ok") ? "true" : "false");
```
Comment (qingwli) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceSender.java]: If false, Ditto

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceAlertChannel.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+        AlertResult alertResult = voiceSender.send();
+        boolean flag = Boolean.parseBoolean(String.valueOf(alertResult.getStatus()));
+        if (flag) {
+            log.info("alert send success");
+            alertResult.setStatus("true");
+            alertResult.setMessage("vocie send success.");
+        } else {
+            alertResult.setMessage("vocie send error.");
+            log.info("alert send error : {}", alertResult.getMessage());
+        }
```
Comment (qingwli) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceAlertChannel.java]: This logic does not very grace, please refer `sendDingTalkMsg` method

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceSender.java (hunk 6 above)
Comment (xujiaqiang) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceSender.java]: completed

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceAlertChannel.java (hunk 8 above)
Comment (xujiaqiang) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceAlertChannel.java]: completed

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceSender.java (hunk 7 above)
Comment (xujiaqiang) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceSender.java]: completed

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceSender.java (hunk 5 above)
Comment (xujiaqiang) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceSender.java]: completed

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceParam.java [hunk 9]
```diff
@@ -17,9 +17,8 @@
 
 package org.apache.dolphinscheduler.plugin.alert.voice;
 
-import lombok.Data;
 
-@Data
+
```
Comment (qingwli) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceParam.java]: Why remove `@Data` ?

Comment (xujiaqiang) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceParam.java]: Change to  getter and setter

Comment (qingwli) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceParam.java]: `@Data` is better

Comment (qingwli) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceParam.java]: @xujiaqiang

Comment (xujiaqiang) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceParam.java]: @qingwli  Changed to @Data

**Review by qingwli — DISMISSED** [review]
LGTM

**Comment by qingwli** [comment]
Please fix UT. Add your new jar license like `DEPENDENCIES`

**Review by davidzollo — DISMISSED** [review]
+1 LGTM

**Comment by zhongjiajie** [comment]
could you format the code via `mvn spotless:apply`

**Pull request closed without merging**